build_scripts/build_collect_diff.sh

```diff
@@ -0,0 +1,157 @@
+    # Generate compiled source files list
+    local json_output_path="$root_path/scripts/change-impact-tools/build_data/compile_commands.json"
+    echo "path check: $(pwd)"
+    python3 scripts/clang-tools/gen_compile_commands.py -o "$json_output_path"
+
+    parse_source_json_file
+    echo "source compiled filelist generated to sourcefile.txt"
+
+    # Generate compiled header files list
+
+    local output_list="$root_path/scripts/change-impact-tools/build_data/headerfile.txt"
+    local output_json="$root_path/scripts/change-impact-tools/build_data/source_dep.json"
+    local dep_path="dependency_file.txt"
+    local python_tool_path="$root_path/scripts/change-impact-tools/build_scripts/parse_dep_list.py"
+
+    python3 "$python_tool_path" "$dep_path" "$output_json" "$output_list"
+    echo "dependency compiled filelist generated to headerfile.txt$"
+
+}
+
+# clean up the working directory
+cleanup_working_directory() {
+    git reset --hard
+    git clean -fdx
+}
+
+# generate diff for build between TAG1 and TAG2
+generate_git_diff() {
+
+    # collect and setup input & output file
+    file_type=${1:-source}
+    local root_build_data_path="$root_path/scripts/change-impact-tools/build_data"
+    local diff_input="$root_build_data_path/sourcefile.txt"
+    local diff_output="$root_build_data_path/filtered_diff_source.txt"
+
+    if [ "$file_type" = "header" ]; then
+        echo "[generate_git_diff] Generating dependency git diff report ..."
+        diff_input="$root_build_data_path/headerfile.txt"
+        diff_output="$root_build_data_path/filtered_diff_header.txt"
+    else
+        echo "[generate_git_diff] Generating source git diff report ..."
+    fi
+    echo "parsing for subsys: $SUBSYS"
+    while IFS= read -r file
+    do
+        if [[ "$file" == $SUBSYS/* ]]; then
+            echo "now generating git diff for $file"
+            if git show "$TAG2:$file" &> /dev/null; then
+                local diff_result
+                echo "$file suitable for parse"
+                diff_result=$(git diff "$TAG1" "$TAG2" -- "$file")
+                if [[ -n "$diff_result" ]]; then
+                    {
+                        echo "Diff for $file"
+                        echo "$diff_result"
+                        echo ""
+                    } >> "$diff_output"
+
+                fi
+            fi
+        fi
+    done < "$diff_input"
+    echo "[generate_git_diff] Git diff report for $file_type files save to compiled_data"
+
+}
+
+
+TAG1="$1"
+TAG2="$2"
+SUBSYS="$3"
+echo "build and collect kernel for subsystem: $SUBSYS"
+
+# Fetch tags from the repository
+git fetch --tags
+echo "Generating source file list for $TAG1"
+git checkout "$TAG1"
+echo "starting to run make olddefconfig"
+make olddefconfig
+echo "finished make olddefconfig"
+
+
+# Preparation before running make
+apply_patch
+install_package_safe
+
+# Build linux kernel
+echo "the current os-linux version: "
+cat /etc/os-release
+
+echo "start running make"
+make 
```

Comment:
I imagine you want the kernel to build to run as quickly as possible. Consider building on all available processors, like this:

```suggestion
make -j $(nproc)
```

Reply:
Change was applied.